components/persistor/lib/knex/db.ts

```diff
@@ -982,6 +990,37 @@ module.exports = function (PersistObjectTemplate) {
         }
     };
 
+    /**
+     * Get the insert sql for the given object
+     *
+     * @param {object} template super type
+     * @returns {string} raw insert sql
+     */
+     PersistObjectTemplate.getInsertScript = function (obj) {
+        var template = obj.__template__;
+        var tableName = this.dealias(template.__table__);
+        var knex = this.getDB(this.getDBAlias(template.__table__)).connection(tableName);
+        const props = template.getProperties();
+        var schema = template.__schema__;
+        var data = {};
+        for (var prop in props) {
+            var defineProperty = props[prop];
+            if (!this._persistProperty(defineProperty) || 
+                (defineProperty.type === Array && defineProperty.of.isObjectTemplate))
+                continue;
+            else if (defineProperty.type && defineProperty.type.__objectTemplate__) {
+                if (!schema || !schema.parents || !schema.parents[prop] || !schema.parents[prop].id)
+                    throw new Error(template.__name__ + '.' + prop + ' is missing a parents schema entry');
+                var foreignKey = (schema.parents && schema.parents[prop]) ? schema.parents[prop].id : prop;
+                data[foreignKey] = obj[prop];
+            }
+            else 
+                data[prop] = obj[prop];
+        }
+
+        return knex.insert(data).toString();
```

Comment:
This just generates as a string correct? should we make it possible to pass the query builder object back?

Reply:
Thought about that approach, but the current usecase is making the consumer write loops and all if we return query builder, we want to track them in a file backup and run all them once.